core/src/main/scala/org/ensime/core/Project.scala

```diff
@@ -49,11 +50,13 @@ class Project(
   private val searchService = new SearchService(config, resolver)
   private val sourceWatcher = new SourceWatcher(config, resolver :: Nil)
   private val reTypecheck = new FileChangeListener {
-    def reTypeCheck(): Unit = self ! AskReTypecheck
-    def fileAdded(f: FileObject): Unit = reTypeCheck()
-    def fileChanged(f: FileObject): Unit = reTypeCheck()
-    def fileRemoved(f: FileObject): Unit = reTypeCheck()
-    override def baseReCreated(f: FileObject): Unit = reTypeCheck()
+    private val reTypeCheck = Debouncer("AskReTypecheck", system.scheduler, 5.seconds, 20.seconds) { () =>
```

Comment:
I'm a bit nervous about non-akka hardcoded time durations because of the slower Windows / OSX build machines that we use. We have used manual dilation (e.g. see `Timing` in utils) where akka doesn't manually manage this. I wonder if perhaps we should do the same here.